query-core/src/main/java/org/infinispan/query/core/impl/MetadataHybridQuery.java

```diff
@@ -9,13 +9,15 @@
 import org.infinispan.objectfilter.ObjectFilter;
 import org.infinispan.objectfilter.impl.syntax.parser.IckleParsingResult;
 import org.infinispan.objectfilter.impl.syntax.parser.projection.ScorePropertyPath;
+import org.infinispan.objectfilter.impl.syntax.parser.projection.VersionPropertyPath;
 import org.infinispan.query.core.stats.impl.LocalQueryStatistics;
 import org.infinispan.query.dsl.Query;
 import org.infinispan.query.dsl.QueryFactory;
 
 public class MetadataHybridQuery<T, S> extends HybridQuery<T, S> {
 
    private final BitSet scoreProjections;
+   private boolean versionProjection;
```

Comment:
I would also recommend making this final. Just need to make a local variable in the constructor and then set it after the for loop.